wordpress-mobile/WordPress-Android
# Intercept wordpress.com post links

Fixes #4698 

Added with this PR is also the ability to invoke the Reader (via deeplinking) without an account (WordPress.com or self-hosted) logged in. This way, the user can benefit from the reader even if the app is merely installed and never logged in.

To test:
- Use some apps (like Hangouts/Skype/Slack/Facebook) and try to open a blog post from a site hosted on wordpress.com without a custom domain mapped. Note, the Facebook app needs altering a setting to disable opening links in its internal browser.
- Notice the picker coming up asking you to select which app to use for that action. WordPress should be on the picker's list.
- Tap on the WordPress app and either select "Use once" or "Always" if asked
- The app should come up and display the post directly inside the Reader
- If it's a link to comments (or reply to comment) the comments view will open directly

Supported URL forms:
- https://*.wordpress.com/{year}/{month}/{day]/{post-slug}
- https://*.wordpress.com/{year}/{month}/{day]/{post-slug}/#comments
- https://*.wordpress.com/{year}/{month}/{day]/{post-slug}/#comment-{id}
- https://*.wordpress.com/{year}/{month}/{day]/{post-slug}/#repond
- https://*.wordpress.com/{year}/{month}/{day]/{post-slug}/?replytocom={id}#respond
- https://*.wordpress.com/{year}/{month}/{day]/{post-slug}/?like_actor={like_actor_id}&like=1
- https://*.wordpress.com/{year}/{month}/{day]/{post-slug}/?like_actor={like_actor_id}&commentid={commentId}&like=1

In case you, dear reviewer, want to test those URLs via the command line, the ADB command is: `adb -s __device serial__ shell am start -W -a android.intent.action.VIEW -d "__URL__"`. But, to mind that the `&`s will probably need to be escaped in that `__URL__`.
